pkg/testing/integration/program.go

```diff
@@ -1570,6 +1570,29 @@ func (pt *ProgramTester) copyTestToTemporaryDirectory() (string, string, error)
 		return "", "", errors.Wrapf(err, "Failed to prepare %v", projdir)
 	}
 
+	// TODO[pulumi/pulumi#5455]: Dynamic providers fail to load when used from multi-lang components.
+	// Until that's been fixed, this environment variable can be set by a test, which results in
+	// a package.json being emitted in the project directory and `yarn install && yarn link @pulumi/pulumi`
+	// being run.
+	// When the underlying issue has been fixed, the use of this environment variable should be removed.
+	if os.Getenv("PULUMI_TEST_YARN_LINK_PULUMI") != "" {
```

Comment:
Went with an env variable to temporarily enable this, instead of exposing an actual option on `ProgramTestOptions`, since this is just a temporary workaround.

Originally, I tried doing this as part of building the tests in the `Makefile` after:

https://github.com/pulumi/pulumi/blob/b4c2a3521c77bc5acd97a90be1bdceb0e906f8c4/Makefile#L71-L72

But our [beloved `CopyFile`](https://github.com/pulumi/pulumi/issues/5230) utility function that we use to copy the test project dir to a temp dir, doesn't handle copying the symlinks that get created inside the `node_modules` dir when `yarn link`ing `@pulumi/pulumi`.